File: tests/playground.c
#include "io.h"

void shadow(int a) {
  if (a > 0) {

    if (a == 1) {
        print_s("1");
    }
    else if (a == 2) {
        print_s("2");
    }
    else if (a == 3) {
        print_s("3");
    }
    else {
        if (a == 4) {
            print_s("4");
        }
        else {
            print_i(a);
        }

    }

  }
}

void main() {
    int a;

    print_s("Get int > ");
    a = read_i();
    shadow(a);
}

File: src/Playground.java


public class Playground {


    public static void main(String[] args) {
        int i = IO.read_i();
        int upper = i >> 16;
        int lower = i & 0x0000FFFF;

        int result = (upper << 16) + lower;

        IO.print_i(upper);
        IO.print_s("/n");
        IO.print_i(lower);
        IO.print_s("/n");
    }

}
